### PR TITLE
[iOS] Swap summary

### PR DESCRIPTION
**Pull request by thangpham26394**

### CODE REVIEW

File: KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  }
+  
+  
+  func fetchRate() {
+    let provider = MoyaProvider<KrytalService>(plugins: [NetworkLoggerPlugin(verbose: true)])
+    provider.requestWithFilter(.getExpectedRate(src: self.swapObject.sourceToken.address.lowercased(), dst: self.swapObject.destToken.address.lowercased(), srcAmount: self.swapObject.sourceAmount.description, hint: self.swapObject.rate.hint, isCaching: true)) { [weak self] result in
+      guard let `self` = self else { return }
+      if case .success(let resp) = result, let json = try? resp.mapJSON() as? JSONDictionary ?? [:], let rate = json["rate"] as? String, let priceImpact = json["priceImpact"] as? Int {
+//        if self.swapObject.rate.rate != rate {
+        self.rootViewController?.rateUpdated(newRate: rate, priceImpact: priceImpact)
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift:171]: ViewModel should not keep reference to ViewController

File: KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/Views/SwapInfoView.swift [hunk 2]
```diff
@@ -48,9 +48,17 @@ class SwapInfoView: BaseXibView {
...
     titleLabel.text = title
     underlineView.isHidden = !underlined
+    iconImageView.isHidden = !shouldShowIcon
+
+
+
+
+
+
+
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/Views/SwapInfoView.swift:61]: Remove redundant lines

File: KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/Views/SwapSummaryViewController.swift [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+    maxGasFeeInfoView.setTitle(title: "Max Network Fee", underlined: true)
+    maxGasFeeInfoView.onTapTitle = { [weak self] in
+      self?.showBottomBannerView(message: Strings.swapTxnMaxFeeInfo, icon: Images.swapInfo)
+    }
+
+    priceImpactInfoView.setTitle(title: "Price Impact", underlined: true)
+    priceImpactInfoView.onTapTitle = { [weak self] in
+      self?.showBottomBannerView(message: Strings.swapPriceImpactInfo, icon: Images.swapInfo)
+    }
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/Views/SwapSummaryViewController.swift:135]: Add event click info title

File: KyberNetwork/KyberNetwork/Screens/CommonViews/Token/SearchToken/SearchTokenViewController.swift [hunk 4]
```diff
@@ -52,7 +53,7 @@ class SearchTokenViewController: KNBaseViewController {
     }
     
     self.viewModel.fetchDataFromAPI(querry: "", orderBy: self.orderBy) {
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/CommonViews/Token/SearchToken/SearchTokenViewController.swift:55]: typo `querry`

File: KyberNetwork/KyberNetwork/Screens/CommonViews/Token/SearchToken/SearchTokenViewController.swift [hunk 5]
```diff
@@ -129,14 +135,14 @@ extension SearchTokenViewController: UITextFieldDelegate {
   
   func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
     timer?.invalidate()
-    timer = Timer.scheduledTimer(timeInterval: 0.5, target: self, selector: #selector(doSearch), userInfo: nil, repeats: false)
+    timer = Timer.scheduledTimer(timeInterval: 0, target: self, selector: #selector(doSearch), userInfo: nil, repeats: false)
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/CommonViews/Token/SearchToken/SearchTokenViewController.swift:138]: time interval is 0

File: KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+  
+  fileprivate func openTransactionStatusPopUp(transaction: InternalHistoryTransaction) {
+    self.internalHistoryTransaction.value = transaction
+  }
+  
+  func sendSignedTransactionDataToNode(data: Data, eip1559Tx: EIP1559Transaction, internalHistoryTransaction: InternalHistoryTransaction) {
+    guard let provider = self.session.externalProvider else {
+      return
+    }
+//    self.navigationController.displayLoading()
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift:236]: Remove unused commented lines

File: KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+
+  func getEstimateGasLimit(txEIP1559: EIP1559Transaction?, tx: SignTransaction?) {
+    if let txEIP1559 = txEIP1559 {
+      KNGeneralProvider.shared.getEstimateGasLimit(eip1559Tx: txEIP1559) { (result) in
+//        self.navigationController.hideLoading()
+        switch result {
+        case .success:
+          let internalHistory = InternalHistoryTransaction(type: .swap, state: .pending, fromSymbol: self.swapObject.sourceToken.symbol, toSymbol: self.swapObject.destToken.symbol, transactionDescription: "\(self.leftAmountString) -> \(self.rightAmountString)", transactionDetailDescription: self.displayEstimatedRate, transactionObj: nil, eip1559Tx: txEIP1559)
+            
+          internalHistory.transactionSuccessDescription = "\(self.leftAmountString) -> \(self.rightAmountString)"
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift:269]: Use →

File: KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift [hunk 8]
```diff
@@ -0,0 +1,395 @@
...
+    case .medium:
+      return KNGasCoordinator.shared.standardPriorityFee
+    case .slow:
+      return KNGasCoordinator.shared.lowPriorityFee
+    case .superFast:
+      return KNGasCoordinator.shared.superFastPriorityFee
+    default: // No need to handle case .custom
+      return KNGasCoordinator.shared.standardPriorityFee
+    }
+  }
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift:393]: Remove this

File: KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/Views/SwapSummaryViewController.swift [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+extension SwapSummaryViewController: SwapProcessPopupDelegate {
+  func swapProcessPopup(_ controller: SwapProcessPopup, action: SwapProcessPopupEvent) {
+    controller.dismiss(animated: true) {
+      switch action {
+      case .openLink(let url):
+        self.navigationController?.openSafari(with: url)
+      case .goToSupport:
+        self.navigationController?.openSafari(with: "https://t.me/KrystalDefi")
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/Views/SwapSummaryViewController.swift:205]: Move this string to a constant

File: KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift (hunk 6 above)
Comment (thangpham26394) [reply KyberNetwork/KyberNetwork/Screens/Tabbars/SwapV2/ViewModels/SwapSummaryViewModel.swift:236]: I will update later when improve loading HUD

**Review by tungnguyen20 — APPROVED** [review]

**Pull request merged**